File: src/main/java/com/example/demo/service/RentalService.java
package com.example.demo.service;

import com.example.demo.model.Rental;
import java.time.LocalDate;
import java.util.List;

public interface RentalService {
    List<Rental> getAllRentals();
    Rental getRentalById(Integer id);
    Rental createRental(Rental rental);
    Rental updateRental(Integer id, Rental rental);
    void deleteRental(Integer id);
    List<Rental> getRentalsByEmail(String email);
    List<Rental> getActiveRentals();
    List<Rental> getRentalsInDateRange(LocalDate startDate, LocalDate endDate);
    List<Rental> getRentalsByCarId(Integer carId);
    List<Rental> getRentalsInDateRangeForCar(Integer carId, LocalDate start, LocalDate end);
    double calculateRentalPrice(Integer carId, LocalDate startDate, LocalDate endDate);
}

File: frontend/src/services/rentalService.js
import axios from 'axios';

const API_URL = '/api/rentals';

export default {
    async getAllRentals() {
        const response = await axios.get(API_URL);
        return response.data;
    },
    
    async getRentalById(id) {
        const response = await axios.get(`${API_URL}/${id}`);
        return response.data;
    },
    
    async createRental(rentalData) {
        const response = await axios.post(API_URL, rentalData);
        return response.data;
    },
    
    async updateRental(id, rentalData) {
        const response = await axios.put(`${API_URL}/${id}`, rentalData);
        return response.data;
    },
    
    async deleteRental(id) {
        await axios.delete(`${API_URL}/${id}`);
    },
    
    async getRentalsByEmail(email) {
        const response = await axios.get(`${API_URL}/email/${email}`);
        return response.data;
    },
    
    async getActiveRentals() {
        const response = await axios.get(`${API_URL}/active`);
        return response.data;
    },
    
    async getRentalsInDateRange(startDate, endDate) {
        const response = await axios.get(`${API_URL}/date-range`, {
            params: { startDate, endDate }
        });
        return response.data;
    },

    async getRentalsInDateRangeForCar(carId, startDate, endDate) {
        const response = await axios.get(`${API_URL}/date-range/car/${carId}`, {
            params: { 
                start: startDate,
                end: endDate 
            }
        });
        return response.data;
    },
    
    async cancelRental(id) {
        const response = await axios.put(`${API_URL}/${id}/status`, { status: 'CANCELLED' });
        return response.data;
    }
};
